dump.c

```diff
@@ -704,6 +704,32 @@ S_opdump_indent(pTHX_ const OP *o, I32 level, UV bar, PerlIO *file,
     va_end(args);
 }
 
+struct OpDumpContext {
+    I32 level;
+    UV bar;
+    PerlIO *file;
+};
+
+void
+Perl_opdump_printf(pTHX_ void *ctx, const char *pat, ...)
```

Comment:
Why not make ctx a pointer to struct OpDumpContext?

OpDumpContext could have an incomplete definition in perl.h

Reply:
Yeah that seems a reasonable idea.

Also while I'm there I realised I didn't actually write any docs for the new function - I can add that.